engine/runtime/src/main/scala/org/enso/compiler/core/IR.scala

```diff
@@ -5808,6 +5809,10 @@ object IR {
       case object InvalidOperatorName extends Reason {
         override def explanation: String = "Invalid operator name."
       }
+
+      case object InvalidForeignDefinition extends Reason {
+        override def explanation: String = "Invalid foreign definition."
```

Comment:
This needs to be more specific. "Invalid foreign definition `foo` is not a supported foreign language", or something like that.